### ISSUE TITLE
Refactoring dual costs

### ISSUE THREAD

**Issue by jeremyomer** [wssuite/NurseScheduler#14]
Here is the code of buildDualCosts:

`DualCosts MasterProblem::buildDualCosts(LiveNurse* pNurse) const { return DualCosts(getShiftsDualValues(pNurse), getStartWorkDualValues(pNurse), getEndWorkDualValues(pNurse), getWorkedWeekendDualValue(pNurse)); }`

My issue is that most of the populate functions do not even make sense in a decomposition by rotation.
Actually, I am even wondering whether we should not try and make the DualCosts class more general. E.g., instead of referring to :

`vector2D<double> workedShiftsCosts_; std::vector<double> startWorkCosts_; std::vector<double> endWorkCosts_; double workedWeekendCost_;`

refer only to dual costs on (day,shift)s and on days:
`dualDays_, dualShifts_`
For the rotation model, build
` dualDays_= startWorkCosts_+endWorkCosts_ + (workedWeekendCost_ if day on week-end)`

**Comment by legraina** [comment]
I won't work, we need the dual to start and end .... To make this general, we need to define a link (that would the constraint) between the master and the subproblem.  Once the link defined, we can automatically populate the dual. To be honest, I think it's too much in our. case.

**Comment by jeremyomer** [comment]
I agree with the necessity to have the linking constraints defined. I'm not closing the issue, because I think we should think about it if we generalize the constraints as in Issue #10
